### PR TITLE
[Crafting] SynthUtils work. Added era dependant margins and rates.

### PR DESCRIPTION
**Pull request by Xaver-DaRed**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

Changes:
- Sanitize code flow a little.
- Correct some auto-generated styling.
- Apply JP wiki information to different HQ tier rewards.
- Adds support for both current retail and era Skill Up Windows.
- Adds support for era dependant rates.

### CODE REVIEW

File: src/map/utils/synthutils.cpp [hunk 1]
```diff
@@ -477,181 +409,179 @@ namespace synthutils
...
-                            }
+            if (random < skillUpChance) // If character skills up
+            {
+                int32 skillUpAmount = 1;
 
-                            if (chance < random)
-                            {
-                                break;
-                            }
+                if (charSkill < 600) // no skill ups over 0.1 happen over level 60
```
Comment (TeoTwawki) [inline src/map/utils/synthutils.cpp]: *normally, without some sort of buff to it

`#finePrintDisclaimer`

**Review by claywar — APPROVED** [review]
Seems reasonable to me.  Anyone with more recent crafting experience want to add their thoughts?

**Comment by TeoTwawki** [comment]
> Seems reasonable to me. Anyone with more recent crafting experience want to add their thoughts?

I thought this was waiting on styling (bracing got removed) but I see now nobody actually left commenting to that effect _On github_ ...

**Comment by Xaver-DaRed** [comment]
I would suggest a squash. Also, we could add this to its own branch as a feature and start making good use of the everything branch again. I do not discard further adjustments down the line, after getting some imput from Canaria peeps, for example.

File: src/map/utils/synthutils.cpp [hunk 2]
```diff
@@ -247,81 +222,60 @@ namespace synthutils
...
-                hqtier           = 0;
+                random    = xirand::GetRandomNumber(1.);        // Random call must be called for each involved skill.
+                hqtier    = 0;                                  // Set HQ Tier to 0 AGAIN. Or else bad things happen.
+                synthDiff = getSynthDifficulty(PChar, skillID); // Get synth difficulty again, for each skill involved.
 
                 if (synthDiff <= 0)
                 {
-                    if (PChar->CraftContainer->getCraftType() == 1)
-                    { // if it's a desynth lower success rate
+                    if (PChar->CraftContainer->getCraftType() == 1) // if it's a desynth lower success rate
```
Comment (claywar) [inline src/map/utils/synthutils.cpp:247]: Completely missed this block before Teo's comment, but braces are required for styling around conditionals

Comment (Xaver-DaRed) [reply src/map/utils/synthutils.cpp:247]: Sure, I'll add the braces back

**Pull request merged**